File: features/operational/contact_us/src/main/java/com/tokopedia/contactus/inboxticket2/view/customview/adapter/OptionItemViewHolder.java
package com.tokopedia.contactus.inboxticket2.view.customview.adapter;

import android.graphics.drawable.GradientDrawable;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.ImageView;

import com.tokopedia.contactus.R;
import com.tokopedia.design.quickfilter.ItemFilterViewHolder;
import com.tokopedia.design.quickfilter.QuickFilterItem;
import com.tokopedia.design.quickfilter.QuickSingleFilterListener;

public class OptionItemViewHolder extends ItemFilterViewHolder {

    private ImageView layoutInside;

    public OptionItemViewHolder(View itemView, QuickSingleFilterListener listener) {
        super(itemView, listener);
        layoutInside =itemView.findViewById(R.id.layout_inside1);
    }
    protected void updateData(QuickFilterItem filterItem) {
        filterName.setText(filterItem.getName());
          layoutInside.setBackgroundResource(R.drawable.check_box_bg);
    }

    protected void updateItemColor(boolean selected) {
        if (selected) {
            layoutInside.setBackgroundResource(R.drawable.checked);
            filterName.setTextColor(ContextCompat.getColor(filterName.getContext(), com.tokopedia.design.R.color.font_black_primary_70));
        } else {
            layoutInside.setBackgroundResource(R.drawable.check_box_bg);
            filterName.setTextColor(ContextCompat.getColor(filterName.getContext(), com.tokopedia.design.R.color.grey_500));
        }
    }


}

File: features/operational/csat_rating/src/main/java/com/tokopedia/csat_rating/adapter/OptionItemViewHolder.kt
package com.tokopedia.csat_rating.adapter

import android.graphics.drawable.GradientDrawable
import android.view.View
import com.tokopedia.abstraction.common.utils.view.MethodChecker
import com.tokopedia.csat_rating.R
import com.tokopedia.csat_rating.quickfilter.ItemFilterViewHolder
import com.tokopedia.csat_rating.quickfilter.QuickFilterItem
import com.tokopedia.csat_rating.quickfilter.QuickSingleFilterListener

open class OptionItemViewHolder(itemView: View, listener: QuickSingleFilterListener?) : ItemFilterViewHolder(itemView, listener) {

    override fun updateData(filterItem: QuickFilterItem) {
        filterName?.text = filterItem.name
        layoutBorder?.setBackgroundResource(R.drawable.rounded_stroke_grey)
        layoutInside?.setBackgroundResource(R.drawable.rounded_stroke_grey)
    }

    override fun updateItemColor(selected: Boolean) {
        val drawableInside = layoutInside?.background?.current?.mutate() as GradientDrawable
        val drawableBorder = layoutBorder?.background?.current?.mutate() as GradientDrawable
        if (selected) {
            layoutInside?.context?.let { drawableBorder.setColor(MethodChecker.getColor(it,
                com.tokopedia.unifyprinciples.R.color.Unify_GN400)) }
            layoutBorder?.context?.let { drawableInside.setColor(MethodChecker.getColor(it,
                R.color.csat_dms_reason_bg)) }
            filterName?.context?.let { filterName?.setTextColor(MethodChecker.getColor(it,
                com.tokopedia.unifyprinciples.R.color.Unify_GN500)) }
        } else {
            layoutInside?.context?.let { drawableBorder.setColor(MethodChecker.getColor(it,
                com.tokopedia.unifyprinciples.R.color.Unify_NN300)) }
            layoutBorder?.context?.let { drawableInside.setColor(MethodChecker.getColor(it,
                com.tokopedia.unifyprinciples.R.color.Unify_NN0)) }
            filterName?.context?.let { filterName?.setTextColor(MethodChecker.getColor(it,
                com.tokopedia.unifyprinciples.R.color.Unify_NN950)) }
        }
    }


}
